Commands/Public/safebooru.js

```diff
@@ -0,0 +1,74 @@
+const ArgParser = require("../../Modules/MessageUtils/Parser");
+const PaginatedEmbed = require("../../Modules/MessageUtils/PaginatedEmbed");
+const fetch = require("chainfetch");
+
+module.exports = async ({ Constants: { Colors, Text, APIs, UserAgent, NSFWEmbed }, client }, { serverDocument }, msg, commandData) => {
+	if (!msg.channel.nsfw) {
+		return msg.send(NSFWEmbed);
+	}
+	if (!msg.suffix) {
+		return msg.sendInvalidUsage(commandData, "You gotta give me somethin' to search for!");
+	}
+
+	let num;
+	const args = ArgParser.parseQuoteArgs(msg.suffix, msg.suffix.includes("|") ? "|" : " ");
+	if (!isNaN(args[args.length - 1])) num = args.splice(-1);
+	else num = serverDocument.config.command_fetch_properties.default_count;
+	const query = args.join(" ");
+
+	if (!num || num > serverDocument.config.command_fetch_properties.max_count) num = serverDocument.config.command_fetch_properties.default_count;
+	else num = parseInt(num);
+
+	const m = await msg.send(Text.THIRD_PARTY_FETCH("We're fetching the requested images"));
+
+	const body = await fetch.get(APIs.SAFEBOORU(query, num)).set({
+		Accepts: "application/json",
+		"User-Agent": UserAgent,
+	}).onlyBody()
+		.catch(() => null);
+
+	if (body && body.length) {
+		const descriptions = [], fields = [], images = [];
+		body.forEach(post => {
+			descriptions.push(`Uploaded by ${post.uploader_name}\nClick [here](https://safebooru.donmai.us/posts/${post.id}) to see the full post.${!post.description || post.description === "" ? "" :
+				`\`\`\`css\n${post.description.split("").slice(0, 1500).join("")
```

Comment:
No reason to split afaik

Reply:
Discord embed limits? Our production console is filled to the brim with errors about 400 errors from embeds with fields/descriptions that are too long. I'm planning on implementing manual validation in GABMessage.send anyways because d.js sucks at error handling, but whenever possible, commands should avoid these issues.

Reply:
You can substring strings without needing to create and join an array though :bloblul:

Reply:
This is the method Vlad used in other NSFW commands. It was simply copied over ^^

Reply:
:athis: